ledger/src/shred/merkle.rs

```diff
@@ -251,32 +287,62 @@ impl ShredCode {
     // proof_size is the number of merkle proof entries.
     fn proof_size(&self) -> Result<u8, Error> {
         match self.common_header.shred_variant {
-            ShredVariant::MerkleCode(proof_size) => Ok(proof_size),
+            ShredVariant::MerkleCode(proof_size, _) => Ok(proof_size),
             _ => Err(Error::InvalidShredVariant),
         }
     }
 
     // Size of buffer embedding erasure codes.
-    fn capacity(proof_size: u8) -> Result<usize, Error> {
+    fn capacity(proof_size: u8, chained: bool) -> Result<usize, Error> {
```

Comment:
So many of these functions seem basically the same between Data/Coding shreds. Would be nice to move them up to Shred mod and switch within the functions where necessary. Obviously outside of scope for this review